phgreynoise: Feature - Update to version 2.1.0

hl-tstrawbridge opened the pull request:
## Release Notes

The 2.1.0 update to phgreynoise adds actions to support two additional API endpoints: RIOT (Rule It OuT) and Community. The actions are titled:
- riot lookup ip
	- Identifies IPs from known benign services and organizations that commonly cause false positives.
- community lookup ip
  - An action requiring at least a free community API key to query IPs in the GreyNoise dataset and retrieve a subset of the IP reputation data returned by the lookup ip and lookup reputation actions. A free API key can be obtained at https://www.greynoise.io/viz/signup

Additionally, the API endpoint used to test connectivity was updated.

## What is the current behavior?
The endpoint used to test connectivity was updated

## What is the new behavior?
Two new actions, riot lookup ip and community lookup ip are included in this update.

gusdadiya-crest commented:
We have the following doubts could you please give your input
(1) Found that while passing the proxy variable in the environment variable(ex: HTTPS_PROXY) with the valid value we are getting the error message like "check_hostname requires server_hostname" in the test connectivity. So we found that this error arises because of the "requests" module added in the "pip_dependencies". Because of that, it's using the latest "requests" module and produces the error. In the latest requests module proxy was just configured to accept HTTP traffic but it's not actually using HTTPS. The phantom pre-installed "requests" module working fine. So while removing the "pip_dependencies" then it's using the phantom "requests" module and test connectivity pass as expected. So we removed the "requests" module from the "pip_dependencies". Could you please check from your end that is all working fine?
Please refer:
https://stackoverflow.com/questions/66642705/why-requests-raise-this-exception-check-hostname-requires-server-hostname
https://github.com/urllib3/urllib3/issues/2075

(2) In the "gnql query" action we found that while passing the "0" value in the size parameter, in the response we got the 10 response value and we check the API behavior which is also the same. So we can do one thing while validating the "size" parameter using "_validate_integer" method, we have restricted the "0" value for the "size" parameter.

(3) In the "community lookup ip" action found that sometimes the custom view was empty. The reason is we are checking the condition "result.data.riot" in the "view_reports.html". In response when the "riot" value is "False"(boolean) this condition becomes false and does not go inside the "Community IP Lookup Results" part. For resolving this issue we removed the "result.data.riot" this condition. So could you please check from your side?

hl-tstrawbridge commented:
> Hey @hl-tstrawbridge,
> We have following doubts could you please give your input
> 
> (1) Found that while passing the proxy variable in the environment variable(ex: HTTPS_PROXY) with the valid value we are getting the error message like "check_hostname requires server_hostname" in the test connectivity. So we found that this error arises because of the "requests" module added in the "pip_dependencies". Because of that, it's using the latest "requests" module and produces the error. In the latest requests module proxy was just configured to accept HTTP traffic but it's not actually using HTTPS. The phantom pre-installed "requests" module working fine. So while removing the "pip_dependencies" then it's using the phantom "requests" module and test connectivity pass as expected. So we removed the "requests" module from the "pip_dependencies". Could you please check from your end that is all working fine?
> 
> Please refer:
> https://stackoverflow.com/questions/66642705/why-requests-raise-this-exception-check-hostname-requires-server-hostname
> [urllib3/urllib3#2075](https://github.com/urllib3/urllib3/issues/2075)
> 
> (2) In the "gnql query" action we found that while passing the "0" value in the size parameter, in the response we got the 10 response value and we check the API behavior which is also the same. So we can do one thing while validating the "size" parameter using "_validate_integer" method, we have restricted the "0" value for the "size" parameter.
> 
> (3) In the "community lookup ip" action found that sometimes the custom view was empty. The reason is we are checking the condition "result.data.riot" in the "view_reports.html". In response when the "riot" value is "False"(boolean) this condition becomes false and does not go inside the "Community IP Lookup Results" part. For resolving this issue we removed the "result.data.riot" this condition. So could you please check from your side?

Hi @gusdadiya-crest,

Thanks for pointing these out. I have tested all three changes on my side and can confirm that everything functions as intended.

hl-tstrawbridge commented on Apps/phgreynoise/greynoise.json:
```diff
@@ -545,9 +585,6 @@
                 {
                     "data_path": "action_result.data.*.code_meaning",
                     "data_type": "string",
-                    "contains": [
```
@mjiyani-crest I looked it over and we're safe to just remove that contains, it's not being used for anything.

mjiyani-crest replied:
cool

gusdadiya-crest commented on Apps/phgreynoise/greynoise.json:
```diff
@@ -155,6 +139,122 @@
+                    "data_path": "action_result.data.*.description",
+                    "data_type": "string",
+                    "example_values": [
+                        "Google's global domain name system (DNS) resolution service."
```
Fixed

gusdadiya-crest commented on Apps/phgreynoise/greynoise.json:
```diff
@@ -155,6 +139,122 @@
+                    "data_path": "action_result.data.*.name",
+                    "data_type": "string",
+                    "example_values": [
+                        "Google Public DNS"
```
Fixed

gusdadiya-crest commented on Apps/phgreynoise/greynoise.json:
```diff
@@ -155,6 +139,122 @@
+                    "data_path": "action_result.data.*.explanation",
+                    "data_type": "string",
+                    "example_values": [
+                        "Public DNS services are used as alternatives to ISP's name servers. You may see devices on your network communicating with Google Public DNS over port 53/TCP or 53/UDP to resolve DNS lookups."
```
Fixed

gusdadiya-crest commented on Apps/phgreynoise/greynoise.json:
```diff
@@ -155,6 +139,122 @@
         },
+        {
+            "action": "riot lookup ip",
+            "description": "Lookup IP using GreyNoise's RIOT (Rule It OuT) Endpoint",
```
Fixed
